src/idpi/grib_decoder.py

```diff
@@ -82,6 +52,171 @@ def _extract_pv(pv):
     }
 
 
+@dc.dataclass
+class _FieldBuffer:
+    dims: tuple[str, ...] | None = None
+    hcoords: dict[str, xr.DataArray] = dc.field(default_factory=dict)
+    metadata: dict[str, typing.Any] = dc.field(default_factory=dict)
+    time_meta: dict[int, dict] = dc.field(default_factory=dict)
+    values: dict[tuple[int, ...], np.ndarray] = dc.field(default_factory=dict)
+
+    def load(self, field):
```

Comment:
This method is part of `_FieldBuffer` class which is intended for internal use in this module only. The `_` prefix is a convention that means that the symbol is supposed to be private. For the third load function, I agree it's redundant and might cause confusion. However, removing it would not be in the scope of this PR.